staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -139,3 +151,74 @@ func TestDecode(t *testing.T) {
 		}
 	}
 }
+
+func TestDecodeFieldNameMatching(t *testing.T) {
+	type s struct {
+		LowerA int `cbor:"a"`
+		UpperB int `cbor:"B"`
+	}
+
+	for _, tc := range []struct {
+		name    string
+		decMode cbor.DecMode
+		in      []byte
+		want    s
+	}{
+		{
+			name:    "case-insensitive match",
```

Comment:
This should return a `cbor.UnknownFieldError` with Decode and succeed but not set LowerA with DecodeLax.